docs/test_framework_integration/junit_4.md

```diff
@@ -0,0 +1,42 @@
+# JUnit 4
+
+## `@Rule`/`@ClassRule` integration
+
+**JUnit4 `@Rule`/`@ClassRule`**: This mode starts the container before your tests and tears it down afterwards.
+
+Add a `@Rule` or `@ClassRule` annotated field to your test class, e.g.:
+
+```java
+public class SimpleMySQLTest {
+    @Rule
+    public MySQLContainer mysql = new MySQLContainer();
+    
+    // [...]
+}
+```
+
+
+## Manually controlling container lifecycle
+
+As an alternative, you can manually start the container in a `@BeforeAll`/`@BeforeEach` annotated method in your tests. Tear down will be done automatically on JVM exit, but you can of course also use an `@AfterAll`/`@AfterEach` annotated method to manually call the `close()` method on your container.
```

Comment:
This page describes JUnit 4, but `@BeforeAll`/`@BeforeEach` are not presented in it. Did you mean `@BeforeClass`/`@AfterClass`?

Reply:
Huh! I just moved this from the old docs, but absolutely yes this is wrong: will fix.